jest.config.ts

```diff
@@ -0,0 +1,85 @@
+import type { Config } from '@jest/types';
```

Comment:
Do we really need to rename these to `.ts`?
This probably means that we first need to build before we can run the tests.

Reply:
They do not need to be built beforehand, Jest can use TS config files and the setup script seems to be handled as it is, as well.

Reply:
Having the config files as TS makes it easier to do typing in them, and helps set them apart from compiler-generated JS files. That was my reasoning behind the change, but maybe it is not enough to justify the changes, actually.